src/js/components/Combination.vue

```diff
@@ -26,6 +28,11 @@ export default {
     components: {
         'combination-canvas': CombinationCanvas,
     },
+    props: {
+        corpus: {
+            type: Object,
```

Comment:
If this is a corpus model, you can put that in and set the type to that model. Vue just does an `instanceof` check under the hood.